src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -239,31 +267,62 @@ export class StudyViewPageStore {
         .map(attribute => attribute.clinicalAttributeId);
 
         //update ids when the chart is added or when the filters is updated
-        if(!this._loadedInitially ||
-            attributeIds.length > this._previousQueriedSampleAttributesId.length ||
+        if(attributeIds.length > this._previousQueriedSampleAttributesId.length ||
             (JSON.stringify(this.filters) !== JSON.stringify(this._previousFilters))){
-
             return attributeIds;
         } else {
             return this._previousQueriedSampleAttributesId
         }
-    }
+    }    
 
     @computed get queryPatientAttributeIds() {
         const attributeIds = this.visibleAttributes
         .filter(attribute => attribute.patientAttribute)
         .map(attribute => attribute.clinicalAttributeId);
 
         //update ids when the chart is added or when the filters is updated
-        if(!this._loadedInitially ||
-            attributeIds.length > this._previousQueriedPatientAttributesId.length ||
+        if(attributeIds.length > this._previousQueriedPatientAttributesId.length ||
             (JSON.stringify(this.filters) !== JSON.stringify(this._previousFilters))){
             return attributeIds;
         } else {
             return this._previousQueriedPatientAttributesId
         }
     }
 
+    @computed get clinicalAttributePromises(): { [id: string]: MobxPromise<ClinicalDataCount[]> } {
```

Comment:
is there a reason this doesn't just use remoteData (factory for mobx promises)?

Reply:
I just require a list of `MobxPromise` for each of the attribute. May I'll change to `remoteData` if needed